design/PerformanceProfiles.md

```diff
@@ -0,0 +1,501 @@
+# Performance Profile
+
+This article describes how to add/update Performance Profiles with REST APIs using curl command.
+Documentation still in progress stay tuned.
+
+## CreatePerformanceProfile
+
+This is quick guide instructions to create performance profile using input JSON as follows. For a more detailed guide,
+see [Create Performance Profile](/design/CreatePerformanceProfile.md)
+
+**Request**
+`POST /createPerformanceProfile`
+
+`curl -H 'Accept: application/json' -X POST --data 'copy paste below JSON' http://<URL>:<PORT>/createPerformanceProfile`
+
+```
+{
+  "name": "resource-optimization-openshift",
+  "profile_version": 1,
+  "slo": {
+    "direction": "minimize",
+    "objective_function": {
+        "function_type": "expression",
+        "expression": "request_sum/request_count"
+    },
+    "function_variables": [
+      {
+        "name": "cpuRequest",
+        "datasource": "prometheus",
+        "value_type": "double",
+        "query": "kube_pod_container_resource_requests{pod=~'$DEPLOYMENT_NAME$-[^-]*-[^-]*$', container='$CONTAINER_NAME$', namespace='$NAMESPACE', resource='cpu', unit='core'}",
+        "aggregation_functions": [
+          {
+            "function": "avg",
+            "query": "avg(kube_pod_container_resource_requests{pod=~\"$DEPLOYMENT_NAME$-[^-]*-[^-]*$\", container=\"$CONTAINER_NAME$\", namespace=\"$NAMESPACE\", resource=\"cpu\", unit=\"core\"})"
+          }
+        ]
+      }
+    ]
+  }
+}
+```
+
+**Response**
+
+```
+{
+    "message": "Performance Profile : <name> created successfully. View all performance profiles at /listPerformanceProfiles",
+    "httpcode": 201,
+    "documentationLink": "",
+    "status": "SUCCESS"
+}
+```
+#### Note: One of query or aggregation_functions is mandatory. Both can be present together.
+
+## Update Performance Profile
+
+Update Performance Profile data using input JSON as follows.
+
+**Request**
+`POST /updatePerformanceProfiles`
```

Comment:
Is this API supported in the code now?
Also this should be `updatePerformanceProfile` (singular)

Reply:
Removed it for now as this API is still in-progress.